### DISCUSSION TITLE
Network v21 Upgrade and Lotus Releases Timeline

### DISCUSSION THREAD

**Discussion by rjan90** [filecoin-project/lotus#11334] [Announcement]
**More details on the nv21 upgrade, codenamed 🍉, can [be found here](https://github.com/filecoin-project/community/discussions/74#discussioncomment-7274188)** 

Subscribe to notifications on https://status.filecoin.io/ for any additional updates!
If you have any concerns or questions, leave a comment in the #fil-lotus-help Filecoin slack channel.

## Lotus releases

You will **need to be on Lotus v1.24.0** or later for the Filecoin network 21 upgrade 🍉. We will be publishing two lotus releases before this upgrade:

- **Lotus v1.24.0:** This mandatory version is based on the last stable release, Lotus v1.23.3, and includes only the consensus-breaking changes for the Network nv21 upgrade. We recommend ecosystem teams like node operators to upgrade your node to this version as soon as the release is tagged.

- **Lotus v1.25.0:** This optional feature release will be based on Lotus v1.23.4, and includes all code changes for the network upgrade, as well as all the feature and improvements that are currently currently in the Lotus v1.23.4-release train.

## Releases and Network Upgrade Timeline
> Time below is in UTC. Some final dates/epochs are still being set.

- [x] 2023-10-12: [Lotus v1.24.0-rc2](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0-rc2) (Note: This release candidate sets the upgrade epochs for calibration, and does not set the upgrade epochs for mainnet.)
- [x] 2023-10-18: [Lotus v1.25.0-rc1](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0-rc1) (Note: This release candidate sets the upgrade epochs for calibration, and does not set the upgrade epochs for mainnet.)
- [x] 2023-10-19T13:00:00Z at epoch `1013134`: Calibration nv21 🍉 upgrade
- [x] 2023-10-31: [Lotus v1.24.0-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0-rc3) (Note: This RC includes the UpgradeWatermelonFix / Calibration patch, but does not set the upgrade epochs for mainnet)
- [x]  2023-10-31: [Lotus v1.25.0-rc2](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0-rc2) (Note: This RC includes the UpgradeWatermelonFix / Calibration patch, but does not set the upgrade epochs for mainnet)
- [x] 2023-11-02: [Lotus v1.24.0-rc4](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0-rc4) (Note: This RC includes the UpgradeWatermelonFix / Calibration patch, and sets the mainnet upgrade epoch)
- [x]  2023-11-02: [Lotus v1.25.0-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0-rc3) (Note: This RC includes the UpgradeWatermelonFix / Calibration patch, and sets the mainnet upgrade epoch)
- [x] 2023-11-08T11:00:00Z at epoch `1070494`: Calibration nv21 🍉 patch
- [x] 2023-11-16: [Lotus v1.24.0-rc6](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0-rc6) (Note: This RC includes the UpgradeWatermelonFix2 / Calibration patch, and updates the mainnet upgrade epoch)
- [x] 2023-11-16: [Lotus v1.25.0-rc5](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0-rc5) (Note: This RC includes the UpgradeWatermelonFix2 / Calibration patch, and updates the mainnet upgrade epoch)
- [x] 2023-11-21T13:00:00Z at epoch `1108174 `: Calibration nv21 🍉 patch2
- [x] 2023-11-22: Lotus v1.24.0: https://github.com/filecoin-project/lotus/releases/tag/v1.24.0
- [x] 2023-11-22: Lotus v1.25.0: https://github.com/filecoin-project/lotus/releases/tag/v1.25.0
- [x] 2023-12-09: Hot fix patch - Lotus v1.24.1: https://github.com/filecoin-project/lotus/releases/tag/v1.24.1
- [x] 2023-12-09: Hot fix patch - Lotus v1.25.1: https://github.com/filecoin-project/lotus/releases/tag/v1.25.1
- [x] 2023-12-12: Mainnet nv21 🍉 upgrade epoch 3469380, 2023-12-12T13:30:00Z

**Comment by jennijuju** [comment]
## 2023-11-02

Heya! Quick updates on nv21 🍉

**Calibration - fix the rotten watermelon**
The implementer teams (Venus, Lotus, Forest) has decided to patch the calibration network, fixing the issue w[here](https://github.com/filecoin-project/builtin-actors/pull/1455) partitions with expired sectors needed to be compacted before they can be moved (more details here). Fortunately, we caught it on the testnet before it's launched on mainnet 🎊 And unfortunately, to fix this bug on calibration needs an upgrade as it is a consensus breaking change. That being said, the calibration network will have an patch upgrade codenamed WatermelonFix at **epoch 1070494, 2023-11-08T11:00:00Z**, and this patch will migrate miner actor from CID `bafk2bzacecnh2ouohmonvebq7uughh4h3ppmg4cjsk74dzxlbbtlcij4xbzxq` to `bafk2bzaced7emkbbnrewv5uvrokxpf5tlm4jslu2jsv77ofw2yqdglg657uie` . The new manifest CID of the actor bundles is `bafy2bzacebl4w5ptfvuw6746w7ev562idkbf5ppq72e6zub22435ws2rukzru`.

We ask all the calibration network operators and storage providers to update to the new release candidates as soon as possible and continue testing! 
- [v1.24.0-rc4](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0-rc4)
- [v1.25.0-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0-rc3)

**Mainnet**
We will publish the final release of Lotus v1.24.0 and Lotus v1.25.0 in approximately one week, to November 8th, in order to accommodate for shipping the patch to the calibration testnet and verifying the fix.

Consequently, we have proposed to the core-devs that the Mainnet upgrade to nv21 to be rescheduled to November 29th, one day later than previously planned. The upgrade is scheduled at epoch **3431940, 2023-11-29T13:30**.

Despite this one week push of the final releases of v1.24.0 and v1.25.0 timeline, we are still providing one more week than the the minimum 2 week gap between final releases and the mainnet upgrade epoch, a total of three weeks for lotus users to update their nodes. 

If you have any questions, feel free to ask in #fil-lotus-help

**Comment by jennijuju** [comment]
## 2023-11-16

### nv21 scope and network upgrade timeline updates

**Details can be found at https://github.com/filecoin-project/community/discussions/74?sort=new#discussioncomment-7591806**

tl'dr:
- FIP0070 is descoped from nv21 due to the design and implementation bug discovered.
- Calibration testnet will perform another patch upgrade to remove that code from the v12 bundle. The patch upgrade is scheduled epoch 1108174, which falls on 2023-11-21T13:00:00Z.
- Mainnet upgrade timeline is pushed accordingly to epoch **3469380 - 2023-12-12T13:30:00Z**.

### Lotus releases timeline

For calibration: both v1.24.0-rc6  and v1.25.0-rc5 have been published, please update your calibration nodes to these releases before epoch 1108174, 2023-11-21T13:00:00Z to avoid being on a fork on the network!

For mainnet: we are aiming to publish final releases of v1.24.0 and v1.25.0 on Nov 23rd, which gives node operators 2.5 weeks to update the node!

Thank you all for the testing and the cooperation on Filecoin network upgrades!

**Comment by rjan90** [comment]
### 2023-11-22

The final versions of v1.24.0 and v1.25.0 has now been released:
- [v1.24.0](https://github.com/filecoin-project/lotus/releases/tag/v1.24.0)
- [v1.25.0](https://github.com/filecoin-project/lotus/releases/tag/v1.25.0)

**Mainnet nv21 upgrade epoch is at `3469380 - 2023-12-12T13:30:00Z`.**
